src/main/java/de/learnlib/ralib/solver/simple/Constraint.java

```diff
@@ -26,8 +26,8 @@
 
 public abstract class Constraint {
 
-	public static final Constraint TRUE = Monomial.TRUE;
-	public static final Constraint FALSE = Polynomial.FALSE;
+	public static final Constraint TRUE = new Monomial(Collections.<IntPair>emptyList(), Collections.<IntPair>emptyList());
+	public static final Constraint FALSE = new Polynomial(Collections.<Monomial>emptySet());
```

Comment:
If you use `Monomial#create` and `Polynomial#fromSet` you can probably keep the respective constructors private. Furthermore, by changing the types to `Monomial TRUE = ...` / `Polynomial FALSE = ...` you save a cast in line 65.

Reply:
Thanks for your review.  I've adopted the suggestion(s) regarding `Monomial`.
Using `Polynomial.fromSet` though causes about one third of the tests to fail; perhaps the method needs to be re-written to not use `FALSE` (or needs some other change) ?

Reply:
I've pushed a commit that manages to use `Polynomial.fromSet` and keep the constructor private.
I am not sure I like declaring `TRUE` as `Monomial` anymore; I think declaring it `Constraint` is cleaner.
But comments and suggestions are welcome.

Reply:
Maybe the `makeConjunction` method can be refactored to work with a `Constraint` as well?

Another completely different idea would be to use the old approach, but instead of exposing the `TRUE` / `FALSE` constraints as constants directly, provide static `Constraint#true` / `Constraint#false`  methods that return the `Monomial.TRUE`/ `Polynomial.FALSE`objects instead. This requires some refactoring of the code that accesses the two constraints but should get rid of the flagged access under initialization as well.

Reply:
Thanks again for your input.  I've spent a bit more than half an hour trying the approaches you suggested but with no success.  In the end, I've decided that the best action is to revert the last commit and leave it at that.

The main purpose of this PR is to introduce spotbugs checks, which of course implies fixing the corresponding warnings.  The current set of changes do these; refactoring and further cleanups of the `Constraint` code base is an orthogonal change which can be done in a separate PR.  I suggest this is approved and merged as is.